src/core/or/dos.c

```diff
@@ -677,6 +677,58 @@ dos_log_heartbeat(void)
   return;
 }
 
+/* Helper to export stats for control port getinfo command. */
```

Comment:
We need to document memory management.
```suggestion
/* Helper to export stats for control port getinfo command.
 * Returns a newly-allocated string. */
```